task8/{UPMessenger}/src/main/java/Tweets.java

```diff
@@ -0,0 +1,61 @@
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+import java.util.*;
+
+
+@WebServlet("/tweets/*")
+public class Tweets extends HttpServlet {
+    private void getPost(String id, HttpServletResponse response) throws IOException {
+        Post post = Posts.get(id);
+        response.getWriter().print((new Gson()).toJson(post));
+    }
+
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        List<String> uriList = Arrays.asList(request.getRequestURI().split("/"));
+        if (uriList.get(2).equals("search")) {
```

Comment:
Are you sure "search" will always go third? Well, in your concrete case this may be so, but it looks generally inobvious, like some "magic constant"
I would write `if (request.getRequestURI().contains("/search")`
Or, to be even more precise, I would test against a RegExp pattern like `"/search\b"` (excludes the case when getRequestURI() contains an unexpected segmen like "searchSomethingElse")